src/common/core/coretree.js

```diff
@@ -8,45 +8,15 @@
 define([
     'common/util/assert',
     'common/util/key',
-    'common/core/tasync'
-], function (ASSERT, GENKEY, TASYNC) {
+    'common/core/tasync',
+    'common/util/random',
+    'common/regexp'
+], function (ASSERT, GENKEY, TASYNC, RANDOM, REGEXP) {
 
     'use strict';
 
-    var HASH_REGEXP = new RegExp('#[0-9a-f]{40}');
-    var isValidHash = function (key) {
-        return typeof key === 'string' && key.length === 41 && HASH_REGEXP.test(key);
-    };
-
-    var MAX_RELID = Math.pow(2, 31);
-    var createRelid = function (data) {
-        ASSERT(data && typeof data === 'object');
-
-        var relid;
-        do {
-            relid = Math.floor(Math.random() * MAX_RELID);
-            // relid = relid.toString();
-        } while (data[relid] !== undefined);
-
-        return '' + relid;
-    };
-
-    //// make relids deterministic
-    //if (false) {
-    //    var nextRelid = 0;
-    //    createRelid = function (data) {
-    //        ASSERT(data && typeof data === 'object');
-    //
-    //        var relid;
-    //        do {
-    //            relid = (nextRelid += -1);
-    //        } while (data[relid] !== undefined);
-    //
-    //        return '' + relid;
-    //    };
-    //}
-
-    var rootCounter = 0;
+    var HASH_REGEXP = new RegExp('#[0-9a-f]{40}'),
```

Comment:
This should be in common regexp...
(You can keep the old one for now there too, call this SHA1 or something like that)